### PR TITLE
Implement decompression of DEFLATE and ADOBE_DEFLATE.

### LINKED ISSUE #42: Missing COMPRESSION_ADOBE_DEFLATE
As suggested by the error message, I am reporting the issue:

```julia
ERROR: Compression COMPRESSION_ADOBE_DEFLATE is not implemented. Please open an issue against TiffImages.jl.
Stacktrace:
  [1] error(::String, ::TiffImages.CompressionType, ::String)
    @ Base ./error.jl:42
  [2] read!(tf::TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}, arr::Matrix{ColorTypes.Gray{FixedPointNumbers.Q0f31}}, #unused#::Val{TiffImages.COMPRESSION_ADOBE_DEFLATE})
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/compression.jl:50
  [3] read!(tf::TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}, arr::Matrix{ColorTypes.Gray{FixedPointNumbers.Q0f31}}, comp::TiffImages.CompressionType)
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/compression.jl:8
  [4] read!(target::Matrix{ColorTypes.Gray{FixedPointNumbers.Q0f31}}, tf::TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}, ifd::TiffImages.IFD{UInt32})
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/ifds.jl:212
  [5] load(tf::TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}, ifds::Vector{TiffImages.IFD{UInt32}}, ::Nothing; verbose::Bool)
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/load.jl:63
  [6] load(tf::TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}; verbose::Bool, mmap::Bool)
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/load.jl:36
  [7] load(io::IOStream; verbose::Bool, mmap::Bool)
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/load.jl:7
  [8] #6
    @ ~/.julia/packages/TiffImages/YTEez/src/load.jl:3 [inlined]
  [9] open(f::TiffImages.var"#6#7"{Bool, Bool}, args::String; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:330
 [10] open
    @ ./io.jl:328 [inlined]
 [11] #load#5
    @ ~/.julia/packages/TiffImages/YTEez/src/load.jl:2 [inlined]
 [12] load(filepath::String)
    @ TiffImages ~/.julia/packages/TiffImages/YTEez/src/load.jl:2
 [13] top-level scope
    @ REPL[4]:1
```

### PR DESCRIPTION
**Pull request by GunnarFarneback**
I'm sure this can be made both nicer and faster but this should at least get the functionality started.

Fixes #42.

### CODE REVIEW

**Comment by tlnagy** [comment]
Awesome! Thanks for the PR. Could you add some tests? 

I believe the `underwater_bmx.tif` in https://github.com/tlnagy/exampletiffs uses Deflate compression.

**Comment by GunnarFarneback** [comment]
I was about to make a test with that image but it doesn't seem to actually be there.

**Comment by tlnagy** [comment]
Oh weird. It is still available at the original link: https://people.math.sc.edu/Burkardt/data/tif/tif.html

**Comment by tlnagy** [comment]
Added `underwater_bmx.tif` to the exampletiffs repo, https://github.com/tlnagy/exampletiffs/commit/0ca553559c925dc3e4e84597e99990d0d83bc7bb

**Comment by GunnarFarneback** [comment]
Testcase added. Feel free to replace the `DEFLATE` test with something nicer or to patch it up properly.

**Comment by GunnarFarneback** [comment]
Btw, you may want to check if the exampletiffs README has the size of `underwater_bmx` correct. I had to switch width and height to get the test to pass.

File: test/runtests.jl [hunk 1]
```diff
@@ -86,6 +86,31 @@ end
...
+    img_cvt = convert(Array{eltype(img), ndims(img)}, img)
+    @test img_cvt === img.data
+
+    # Dirty patch the TIFF_COMPRESSION tag from
+    # COMPRESSION_ADOBE_DEFLATE (8) to COMPRESSION_DEFLATE (32946).
+    # Those are in fact exactly the same so nothing else needs to be
+    # adjusted.
+    data = read(filepath)
+    @assert data[1063947:1063948] == [0x08, 0x00]
+    data[1063947:1063948] .= reinterpret(UInt8, [UInt16(32946)])
```
Comment (tlnagy) [inline test/runtests.jl:105]: Clever solution to test the other tag. I think that'll work.

**Comment by kimikage** [comment]
BTW, the `[compat]` entry for `Inflate.jl` is missing.

**Comment by tlnagy** [comment]
Thanks for the heads up @kimikage, added in #52

**Pull request merged**